### PR TITLE
Refactor C interop

### LINKED ISSUE #3894: Fix FFI example fails to find dependent package `test_utils`
### Page URL

https://dart.dev/guides/libraries/c-interop#building-and-running

### Page source

_No response_

### Describe the problem

```
$ dart pub get
Resolving dependencies... 
Because hello_world_ffi depends on test_utils from path which doesn't exist (could
  not find package test_utils at "../test_utils"), version solving failed.
```

### Expected fix

I expect "an example of building the dynamic library and executing the Dart app" to work when followed.

The "hello_world example has the following files" does not mention "../test_utils"

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by atsansone**
Fixes #3894
Expands on this issue by updating links to code examples, updating style

### CODE REVIEW

File: src/content/interop/c-interop.md [hunk 1]
```diff
@@ -79,269 +86,296 @@ Hello World
...
 
 [codesign]: {{site.apple-dev}}/library/content/documentation/Security/Conceptual/CodeSigningGuide/Introduction/Introduction.html
-  
 
-### Using dart:ffi
+#### Leverage dart:ffi
 
-The [`hello.dart` file]({{hw}}/hello.dart)
-illustrates the steps for using `dart:ffi` to call a C function:
+To learn how to call a C function using the `dart:ffi` package,
```
Comment (dcharkes) [inline src/content/interop/c-interop.md:97]: `dart:` is not a package, `package:` is a package. Did you mean to say library?

File: src/content/interop/c-interop.md [hunk 2]
```diff
@@ -79,269 +86,296 @@ Hello World
...
-
-* Flutter `dart:ffi` pages: [Android][android], [iOS][ios], and [macOS][macos] 
+   ```dart
+   hello();
+   ```
+
+Once you understand the `hello_world` example,
+consult the [other `dart:ffi` examples](#download-example-files).
+
+## Bundle and load C libraries
```
Comment (dcharkes) [inline src/content/interop/c-interop.md:167]: I suppose we mostly want to reference what is non-experimental here.

The work on native assets (which is currently an experiment in Dart and Flutter), will make bundling / distributing native code platform agnostic.

* https://github.com/dart-lang/sdk/issues/50565
* https://github.com/flutter/flutter/issues/129757

Edit: Ah we already have a section for native assets, maybe link it here.

File: src/content/interop/c-interop.md [hunk 3]
```diff
@@ -79,269 +86,296 @@ Hello World
...
+This feature should simplify how Dart packages depend on and use native code.
+Native Assets should provide the following benefits:
 
-* It builds (if needed) the native code or
-  obtains the binaries using a package's `build.dart` script.
-* It bundles the native [`Asset`][] reported by the `build.dart` script.
-* It makes the native assets available at runtime through
+* Build the native code or obtains the binaries
+  using a package's `build.dart` script.
+* Bundle the native [`Asset`][] that the `build.dart` script reports.
```
Comment (dcharkes) [inline src/content/interop/c-interop.md:277]: The new terminology is build hook, `build.dart` hook, or `hook/build.dart`. The hook was moved inside the `hook/` directory, and we don't use "script" anymore.

File: src/content/interop/c-interop.md [hunk 4]
```diff
@@ -79,269 +86,296 @@ Hello World
...
-API documentation for the `build.dart` script is available
-on the [`package:native_assets_cli` API reference][].
+### Review Native Asset API documentation
+
+API documentation can be found for the following packages:
 
-### Experiment opt-in
+* To learn about native assets in Dart FFI,
+  consult the `dart:ffi` API reference for [`Native`][] and [`DefaultAsset`][].
+* To learn about the `build.dart` script,
```
Comment (dcharkes) [inline src/content/interop/c-interop.md:320]: script -> hook

File: src/content/interop/c-interop.md [hunk 5]
```diff
@@ -79,269 +86,296 @@ Hello World
...
 
+[`src/native_add_library.c`]: {{native-assets}}/src/native_add_library.c
+[`lib/native_add_library.dart`]: {{native-assets}}/lib/native_add_library.dart
+[`test/native_add_library_test.dart`]: {{native-assets}}/test/native_add_library_test.dart
+[`hook/build.dart`]: {{native-assets}}/hook/build.dart
+
 When a Dart or Flutter project depends on `package:native_add_library`,
-the `build.dart` script will automatically be
-invoked on `run`, `build`, and `test` commands.
+it invokes the `build.dart` script on `run`, `build`, and `test` commands.
```
Comment (dcharkes) [inline src/content/interop/c-interop.md:311]: `build.dart` -> `hook/build.dart`

File: src/content/interop/c-interop.md [hunk 6]
```diff
@@ -79,269 +86,296 @@ Hello World
...
+------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [`src/native_add_library.c`][]          | The C file containing the code for `add`.                                                                                                                                      |
+| [`lib/native_add_library.dart`][]       | The Dart file that invokes the C function `add` in asset `package:native_add_library/native_add_library.dart` through FFI. (Note that _asset id_ defaults to the library uri.) |
+| [`test/native_add_library_test.dart`][] | A Dart test using the native code.                                                                                                                                             |
+| [`hook/build.dart`][]                   | A script for compiling `src/native_add_library.c` and declaring the compiled asset with  id `package:native_add_library/native_add_library.dart`.                              |
```
Comment (dcharkes) [inline src/content/interop/c-interop.md:301]: script -> hook

**Review by dcharkes — APPROVED** [review]
Thanks @atsansone!

File: src/content/interop/c-interop.md (hunk 2 above)
Comment (atsansone) [reply src/content/interop/c-interop.md:167]: Already done on line [264](https://github.com/dart-lang/site-www/pull/5881/files#diff-32de1796331bba2e5ab22ab36136524d890f3de00d3b6d1a16ceb32c55f92407R264)

**Pull request merged**